src/main/java/org/commcare/formplayer/auth/CommCareSessionAuthFilter.java

```diff
@@ -0,0 +1,60 @@
+package org.commcare.formplayer.auth;
+
+import lombok.extern.apachecommons.CommonsLog;
+import org.commcare.formplayer.util.Constants;
+import org.commcare.formplayer.util.RequestUtils;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.web.authentication.preauth.AbstractPreAuthenticatedProcessingFilter;
+
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Auth filter that extracts the user details from the request (username, domain) along
+ * with the session ID from the cookie.
+ *
+ * These are then used to authenticate the user using the HQUserDetailsService
+ */
+@CommonsLog
+public class CommCareSessionAuthFilter extends AbstractPreAuthenticatedProcessingFilter {
+
+    @Override
+    protected Object getPreAuthenticatedCredentials(HttpServletRequest request) {
+        if(request.getCookies() != null) {
+            for (Cookie cookie : request.getCookies()) {
+                if(Constants.POSTGRES_DJANGO_SESSION_ID.equals(cookie.getName())){
+                    return cookie.getValue();
+                }
+            }
+        }
+        return null;
+    }
+
+    @Override
+    protected Object getPreAuthenticatedPrincipal(HttpServletRequest request) {
+        if (HttpMethod.valueOf(request.getMethod()) == HttpMethod.POST) {
+            try {
+                JSONObject data = RequestUtils.getPostData(request);
+                String username = getUsername(data);
+                String domain = data.getString("domain");
+                if (username != null && domain != null) {
+                    return new UserDomainPreAuthPrincipal(username, domain);
+                }
+            } catch (Exception e){
+                return null;  // TODO: should sentry capture this?
```

Comment:
It definitely seems like this should be a typed error rather than a null response. I shudder trying to imagine debugging this.

Reply:
the null response is intentional and an expected return value if the method was unable to extract the necessary information from the request (e.g. if the body wasn't JSON or didn't contain the `username` or `domain` key).

In the case of returning a `null` this filter will not put an authenticated user into the security context and ultimately the request will be rejected (if it is requesting a URL that requires auth) unless there is another filter in the chain that can perform the auth.

I add logging for the error though: df6f00ae